apps/webapp/src/modules/balances/components/BalancesAssets.tsx

```
@@ -8,22 +8,34 @@ import { LoadingErrorWrapper } from '@/modules/ui/components/LoadingErrorWrapper
+export function BalancesAssets({ chainIds }: BalancesAssetsProps) {
   const { address } = useAccount();
-  const chainId = useChainId();
+  const currentChainId = useChainId();
+  const chainsToQuery = chainIds ?? [currentChainId];
   const { data: pricesData, isLoading: pricesIsLoading, error: pricesError } = usePrices();
 
-  const tokens = defaultConfig.balancesTokenList[chainId];
+  // Create an object mapping chainIds to their tokens
+  const chainTokenMap = chainsToQuery.reduce<Record<number, TokenItem[]>>(
+    (acc, chainId) => ({
+      ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)